unison-share-api/src/Unison/Server/Backend.hs

```diff
@@ -322,25 +324,34 @@ findShallowReadmeInBranchAndRender ::
   Rt.Runtime Symbol ->
   Codebase IO Symbol Ann ->
   NamesWithHistory ->
-  Branch IO ->
+  V2Branch.Branch m ->
   Backend IO (Maybe Doc.Doc)
 findShallowReadmeInBranchAndRender width runtime codebase printNames namespaceBranch =
   let ppe hqLen = PPE.fromNamesDecl hqLen printNames
 
+      renderReadme :: PPE.PrettyPrintEnvDecl -> V2.Referent -> IO Doc.Doc
       renderReadme ppe r = do
-        (_, _, doc) <- renderDoc ppe width runtime codebase (Referent.toReference r)
+        let docReference = case r of
+              -- This shouldn't ever happen unless someone puts a non-doc as their readme.
+              V2.Con ref _conId -> Cv.reference2to1 ref
+              V2.Ref ref -> Cv.reference2to1 ref
```

Comment:
What's happening here? :)

Reply:
This is maintaining the existing behaviour, but I think the existing behaviour is non-sensicle in this case. I should probably just return Nothing if it's a Con and fall-through to any other (potentially valid) readmes; what do you think of that change?

Reply:
I cleaned it up into a safer and more correct impl :)